File: app/src/main/java/me/imzack/app/ender/injector/module/TypeSearchPresenterModule.java
package me.imzack.app.ender.injector.module;

import me.imzack.app.ender.view.contract.TypeSearchViewContract;

import dagger.Module;
import dagger.Provides;

@Module
public class TypeSearchPresenterModule {

    private final TypeSearchViewContract mTypeSearchViewContract;

    public TypeSearchPresenterModule(TypeSearchViewContract typeSearchViewContract) {
        mTypeSearchViewContract = typeSearchViewContract;
    }

    @Provides
    TypeSearchViewContract provideTypeSearchViewContract() {
        return mTypeSearchViewContract;
    }
}

File: app/src/main/java/net/zackzhang/app/end/injector/module/TypeSearchPresenterModule.kt
package net.zackzhang.app.end.injector.module

import dagger.Module
import dagger.Provides
import net.zackzhang.app.end.view.contract.TypeSearchViewContract

@Module
class TypeSearchPresenterModule(private val mTypeSearchViewContract: TypeSearchViewContract) {

    @Provides
    fun provideTypeSearchViewContract() = mTypeSearchViewContract
}
